### ISSUE TITLE
Can't delete a local user.

### ISSUE THREAD

**Issue by UnMenorqui** [owncloud/core#27563]
### Steps to reproduce
1. delete a local user on owncloud
2. refresh the user's page
3. the deleted user, keeps existing

### Expected behaviour
The user should disappear

### Actual behaviour
The user keeps existing

### Server configuration
**Operating system**: Ubuntu 16.04 LTS

**Web server:**

Apache

**Database:**
Mysql

**PHP version:**
5.4
**ownCloud version:** (see ownCloud admin page)
9.1.4
**Updated from an older ownCloud or fresh install:**
No

**Comment by PVince81** [comment]
logs?

**Comment by UnMenorqui** [comment]
[owncloud.zip](https://github.com/owncloud/core/files/893142/owncloud.zip)

It's a long log file, so i upload it compressed. 
I've been looking for some 'delete' sentence, but the ones i've found aren't about users.

**Comment by kdslkdsaldsal** [comment]
@UnMenorqui I don't think that anyone is able to find the needed info in a ~70MB plaintext logfile 😉 

Please clear the owncloud.log and then delete the user to only get the relevant logentries.

You might also need to have a look at your webservers logfiles which could provide additional info

**Comment by UnMenorqui** [comment]
Hi again!

Now i've found what it's doing owncloud, if you look at the 
[owncloud.txt](https://github.com/owncloud/core/files/895397/owncloud.txt) at line 12, there's query wondering to delete the 'i1' user.

Let me explain a bit the situation, i've got an LDAP server connected to Owncloud, and also some local users we can't have on our DB. So, i started testing things and see that after deleting a local user and refreshing the page, keeps existing. Even, when i'm deleting his files from the system(Server).

Have you guys any idea of what's happening? or if it's something related to LDAP server?

Thanks!

**Comment by kdslkdsaldsal** [comment]
@UnMenorqui This is a bug caused by the 3rdpary app "Reader (ebook reader)" : https://github.com/Yetangitu/owncloud-apps/issues/43

**Comment by UnMenorqui** [comment]
Thank you very much! That solved my problem.

**Comment by Yetangitu** [comment]
The question here is why it doesn't work with MySQL when it does with PostfreSQL (which I use). There's a discussion at https://github.com/Yetangitu/owncloud-apps/issues/43#issuecomment-291617728 which mentions the hook used in Reader. This hook is more or less identical to the example given in the API documentation with one change: the setParameter() call seems to be misplaced in the documentation, it either misses a leading `$qb` or there is a superfluous `;` between the two. I used the latter notation in Reader - `$qb->(blahblahblah)->setParameter(blahblah)` - which works on my (PostgreSQL-based) installation. I assumed the query builder to be database-agnostic and for it to work across the board but maybe it doesn't?

I posted an experimental patch to the thread mentioned above but have not heard back from the user. I don't have an installation with MySQL or MariaDB handy but if I don't get any feedback I'll have to rig one up to be able to test these changes. If the query builder is not database-agnostic I consider this to be a bug in OC/NC.
